### ISSUE TITLE
About the COCO data file path

### ISSUE THREAD

**Issue by dreamwish1998** [GuangxingHan/QA-FewDet#3]
What should the file path under datasets/coco/ be, for example, whether the train2014 and val2014 data folders are separate or together (trainval2014)? Thanks!

**Comment by GuangxingHan** [comment]
We followed the previous work (TFA, ICML 2020) to prepare the dataset (https://github.com/ucbdrive/few-shot-object-detection/tree/master/datasets). Please see below for an overview of the coco folder.
```
coco/
  annotations/
    trainvalno5k.json
    instances_val2014.json     # same as the 5k.json
    ...  # few-shot annotations
  trainval2014/
    # the combination of train2014 and val2014
  val2014/
    # the original val2014
  ... # few-shot training examples
```

**Comment by dreamwish1998** [comment]
> 我们按照之前的工作（TFA、ICML 2020）准备数据集（https://github.com/ucbdrive/few-shot-object-detection/tree/master/datasets）。请参阅下面的 coco 文件夹概述。
> 
> ```
> coco/
>   annotations/
>     trainvalno5k.json
>     instances_val2014.json     # same as the 5k.json
>     ...  # few-shot training examples
>   trainval2014/
>     # the combination of train2014 and val2014
>   val2014/
>     # the original val2014
> ```

I got it, thanks！

**Comment by dreamwish1998** [comment]
One more question, after running the data processing program in the datasets/coco or datasets/pascal_voc folder, will the original coco or VOC image data still be called by the training or test code? and can they be deleted from the folder? thank you very much!

**Comment by GuangxingHan** [comment]
No, they should not be deleted. We do not save the original images anywhere else with our scripts.

**Comment by dreamwish1998** [comment]
Thanks, is it possible for you to post the pre-trained weights of the base stage (meta-train)? Thank you very much!

**Comment by GuangxingHan** [comment]
Sure. Please see this [link](https://drive.google.com/file/d/1uYOwK7vvymFd7gwg2TMQ7PLK1Q-oLrAz/view?usp=sharing) for  our meta-trained model on the MSCOCO dataset.

**Comment by dreamwish1998** [comment]
If I want to see whether I reproduce the paper's experimental results, what fields of which files should I see? Thanks!!!

**Comment by GuangxingHan** [comment]
Please run the scripts in this repo with both meta-learning and fine-tuning. It will automatically run the evaluation and report the testing results at the end of log for each shot. 

Could you please let me know which step are you at?

**Comment by dreamwish1998** [comment]
> Thanks for the heads up, the code I'm running right now is in the evaluation stage and not over yet. Also, how should I set other hyperparameters when I can't set the batch size to 8 and set it to 4 for now? such as learning rate. Thanks!

**Comment by GuangxingHan** [comment]
I have not tried this before, but you can try different values for the learning rate, like keeping the same value or decreasing the value a little bit.

**Comment by dreamwish1998** [comment]
When the few-shot finetune stage is performed on the COCO dataset, where does the field "ROI_BOX_HEAD:GAMMA_NUM: 2" in the config file work, and what does it mean? Why do I get an error involving this field when I run the few-shot finetune script? Thank you very much for being so patient!

**Comment by GuangxingHan** [comment]
This has been removed from the config files. Thanks for the feedback.

**Comment by dreamwish1998** [comment]
How can I visualize the results of training or inference on the image in this project? Thank you very much!

**Comment by GuangxingHan** [comment]
Can you be more specific what do you want to visualize?
For example, you can try to visualize the constructed graph following the code/paper.

**Comment by dreamwish1998** [comment]
> Can you be more specific what do you want to visualize? For example, you can try to visualize the constructed graph following the code/paper.

I currently want to observe the model's inference results, such as which samples are inferring well. So I want to feed back the final output to the original image for final visualization, specifically in the form of a box. Could you provide some guidance, thanks a lot!

**Comment by GuangxingHan** [comment]
I see. You can visualize the final predicted boxes [here](https://github.com/GuangxingHan/QA-FewDet/blob/main/QA_FewDet/modeling/fsod/fsod_rcnn.py#L874) using the visualization function [here](https://github.com/GuangxingHan/QA-FewDet/blob/main/QA_FewDet/modeling/fsod/fsod_rcnn.py#L113)

**Comment by yomik-js** [comment]
> hello, if I want to visualize the constructed graph, what should I do? I cannot find the code/paper folder.

**Comment by yomik-js** [comment]
> Can you be more specific what do you want to visualize? For example, you can try to visualize the constructed graph following the code/paper.

hello, if I want to visualize the constructed graph, what should I do? I cannot find the folder code/paper.

**Comment by GuangxingHan** [comment]
You can follow the code here [ss_edge](https://github.com/GuangxingHan/QA-FewDet/blob/main/QA_FewDet/modeling/fsod/fsod_rcnn.py#L240) and [_process_per_class](https://github.com/GuangxingHan/QA-FewDet/blob/main/QA_FewDet/modeling/fsod/fsod_rcnn.py#L197), which define the inter-class and intra-class graphs respectively. You can try to visualize the graph with the nodes and edge weights defined in the above functions.
